Mathlib/CategoryTheory/Limits/Shapes/Preorder/HasIterationOfShape.lean

```diff
@@ -20,7 +22,7 @@ universe w v v' u u'
 
 namespace CategoryTheory.Limits
 
-variable (J : Type w) [Preorder J] (C : Type u) [Category.{v} C]
+variable (J : Type w) [LinearOrder J] (C : Type u) [Category.{v} C]
```

Comment:
It seems that some definitions like `HasIterationOfShape` get restricted more through this change. Is that intentional? And should the docstring below be changed from "a preordered type" to "a linearly ordered type"?

Reply:
Yes, the notion is interesting only for well-ordered types.